tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -6666,12 +6670,45 @@ Status ConvertGraphDefToEngine(
   TF_RETURN_IF_ERROR(statusor.status());
   std::unique_ptr<Converter> converter = std::move(statusor.ValueOrDie());
 
+  GraphDef graph = gdef;
+  if (cluster != nullptr) {
+    bool apply_layout_optim;
+    Status status =
+        ReadBoolFromEnvVar("TF_TRT_ENABLE_LAYOUT_OPTIMIZER",
+                           /*default_value=*/true, &apply_layout_optim);
+    if (!status.ok()) {
+      LOG(ERROR) << status;
+    }
+    if (apply_layout_optim) {
+      tensorflow::grappler::GrapplerItem grappler_item;
+      grappler_item.graph = gdef;
+      // TensorRT API requires the input for convolution to be in NCHW.
+      tensorflow::grappler::GenericLayoutOptimizer layout_optimizer("NCHW");
+      TF_RETURN_IF_ERROR(layout_optimizer.Optimize(
+        cluster, grappler_item, &graph));
+
+      grappler_item.graph = graph;
+
+      tensorflow::grappler::ConstantFolding const_optimizer(nullptr,
+                            /*disable_compressed_tensor_optimization=*/false,
+                            /*fold_quantization_emulation=*/false);
+      TF_RETURN_IF_ERROR(const_optimizer.Optimize(
+        cluster, grappler_item, &graph));
+
+      // The optimizers may break the topological order
+      // so we need these steps to restore it
+      Graph g(OpRegistry::Global());
+      TF_CHECK_OK(
```

Comment:
Let's change the error checking here (`TF_CHECK_OK` would abort the process in case of error, which is unnecessary).
```suggestion
      TF_RETURN_IF_ERROR(
```